core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1402,6 +1404,8 @@ class ReplicaManager(val config: KafkaConfig,
                     stateChangeLogger.info(s"Updating log for $topicPartition to assign topic ID " +
                       s"$topicId from LeaderAndIsr request from controller $controllerId with correlation " +
                       s"id $correlationId epoch $controllerEpoch")
+                    if (partitionState.leader != localBrokerId)
+                      topicIdUpdatePartitions.put(partition, partitionState)
```

Comment:
nit: Should we rename `topicIdUpdatePartitions` to indicate that it is only about the followers?